closed: "What's new in Svelte" December newsletter (#9693)
sveltejs/svelte: documentation/blog/2023-12-01-whats-new-in-svelte-december-2023.md

---
title: "What's new in Svelte: December 2023"

> description: 'Svelte 5.0 announced at Svelte Summit Fall 2023'

Reviewer: This sounds like we announced the final release
```suggestion
description: 'Svelte 5 preview announced at Svelte Summit Fall 2023'
```

author: Dani Sandoval
authorURL: https://dreamindani.com
---

> Svelte Summit Fall 2023 was epic! Featuring talks from across the ecosystem and the launch of the beta version of Svelte 5.0! Individual videos for each talk can be found in [this playlist](https://www.youtube.com/watch?v=RT0aemVpvVE&list=PL8bMgX1kyZTh_wu-CruPkxk6sMbXXluJK), but the whole talk can be found [here](https://www.youtube.com/watch?v=pTgIx-ucMsY) (timestamps below):

Reviewer: We debated elsewhere how exactly to frame this, and we settled on unstable preview
```suggestion
Svelte Summit Fall 2023 was epic! Featuring talks from across the ecosystem and the launch of the (unstable!) preview version of Svelte 5! Individual videos for each talk can be found in [this playlist](https://www.youtube.com/watch?v=RT0aemVpvVE&list=PL8bMgX1kyZTh_wu-CruPkxk6sMbXXluJK), but the whole talk can be found [here](https://www.youtube.com/watch?v=pTgIx-ucMsY) (timestamps below):
```

- [0:00](https://www.youtube.com/watch?v=pTgIx-ucMsY&t=0s) Hype music!
- [18:20](https://www.youtube.com/watch?v=pTgIx-ucMsY&t=1100s) Exploring Svelte DevTools - Ignatius B (4 minutes)
- [25:27](https://www.youtube.com/watch?v=pTgIx-ucMsY&t=1527s) How does SvelteKit fare as a SPA Framework - Henry Lie (25 minutes)

> - [53:00](https://www.youtube.com/watch?v=pTgIx-ucMsY&t=3180s) enhanced:img - Ben McCann (6 minutes)

Reviewer: I think this one deserves a callout in the paragraph above 😄

Author: too bad! ;)

- [1:03:44](https://www.youtube.com/watch?v=pTgIx-ucMsY&t=3824s) inlang-paraglide JS for SvelteKit i18n - Samuel Stroschein (10 minutes)
- [1:17:32](https://www.youtube.com/watch?v=pTgIx-ucMsY&t=4652s) How Svelte & GraphQL plays well together - Jean-Yves Couet (31 minutes)
- [1:53:10](https://www.youtube.com/watch?v=pTgIx-ucMsY&t=6790s) svelte-ecosystem-ci - Dominik G (10 minutes)
- [2:23:06](https://www.youtube.com/watch?v=pTgIx-ucMsY&t=8586s) Svelte for Creative Development - Steven Stavrakis (10 minutes)
- [2:35:33](https://www.youtube.com/watch?v=pTgIx-ucMsY&t=9333s) Translating the docs - Romain I'Ourson (11 minutes)
- [2:49:48](https://www.youtube.com/watch?v=pTgIx-ucMsY&t=10188s) ENHANCE! - Paolo Ricciuti (25 minutes)
- [3:18:25](https://www.youtube.com/watch?v=pTgIx-ucMsY&t=11905s) Accessibility tips with Svelte solutions - Enrico Sacchetti (25 minutes)
- [3:49:11](https://www.youtube.com/watch?v=pTgIx-ucMsY&t=13751s) svelte@next - Rich Harris (18 minutes)

## What's new in Svelte, SvelteKit & Language Tools

> [Svelte 5.0](https://svelte-5-preview.vercel.app/docs/introduction) is now available at [svelte@next](https://www.npmjs.com/package/svelte?activeTab=versions) while Svelte 4's current version (`@latest`) is `4.2.7`.

Reviewer: ```suggestion
[Svelte 5 preview](https://svelte-5-preview.vercel.app/docs/introduction) is now available at [svelte@next](https://www.npmjs.com/package/svelte?activeTab=versions) while Svelte 4's current version (`@latest`) is `4.2.7`.
```

For all the release notes going forward, check out [the CHANGELOG on main](https://github.com/sveltejs/svelte/blob/main/packages/svelte/CHANGELOG.md). For the highlights, take a look below!

- TypeScript is now supported natively via acorn-typescript (**5.0.0-next.9**, [#9482](https://github.com/sveltejs/svelte/pull/9482))
- The new `$effect.active` rune returns a boolean to indicate if an effect is active. This lets you set up subscriptions when a value is read (in an effect, or in the template) but also read values without creating subscriptions (**5.0.0-next.10**, [Docs](https://svelte-5-preview.vercel.app/docs/runes#effect-active), [#9591](https://github.com/sveltejs/svelte/pull/9591))
- The new `$effect.root` rune creates a non-tracked scope that doesn't auto-cleanup. This is useful for nested effects that you want to manually control (**5.0.0-next.14**, **Docs**, [#9638](https://github.com/sveltejs/svelte/pull/9638))
- The latest version of language tools adds best-effort fallback typings to `$props()` rune (**extensions-108.1.0**)

> - Language tools better support the `@render` tag by using the `Snippet` type (**extensions-108.0.0**)

Reviewer: ```suggestion
- Language tools better supports the `@render` tag by using the `Snippet` type (**extensions-108.0.0**)
```

Reviewer 2: It was correct initially; my family (singular) supportS me, and my friends (plural) also support me.

---
title: "What's new in Svelte: December 2023"
description: 'Svelte 5 preview announced at Svelte Summit Fall 2023'
author: Dani Sandoval
authorURL: https://dreamindani.com
---

> Svelte Summit Fall 2023 was epic! Featuring talks from across the ecosystem and the launch of the beta version of Svelte 5.0! Individual videos for each talk can be found in [this playlist](https://www.youtube.com/watch?v=RT0aemVpvVE&list=PL8bMgX1kyZTh_wu-CruPkxk6sMbXXluJK), but the whole talk can be found [here](https://www.youtube.com/watch?v=pTgIx-ucMsY) (timestamps below):

Reviewer 3: beta might be more stable than it is :grin: perhaps also mention the other launches?
```suggestion
Svelte Summit Fall 2023 was epic! Featuring talks from across the ecosystem and the launches from the core team of new Svelte Dev Tools, image optimization, and a preview version of Svelte 5.0! Individual videos for each talk can be found in [this playlist](https://www.youtube.com/watch?v=RT0aemVpvVE&list=PL8bMgX1kyZTh_wu-CruPkxk6sMbXXluJK), but the whole talk can be found [here](https://www.youtube.com/watch?v=pTgIx-ucMsY) (timestamps below):
```

[Svelte 5 preview](https://svelte-5-preview.vercel.app/docs/introduction) is now available at [svelte@next](https://www.npmjs.com/package/svelte?activeTab=versions) while Svelte 4's current version (`@latest`) is `4.2.7`.

For all the release notes going forward, check out [the CHANGELOG on main](https://github.com/sveltejs/svelte/blob/main/packages/svelte/CHANGELOG.md). For the highlights, take a look below!

- TypeScript is now supported natively via acorn-typescript (**5.0.0-next.9**, [#9482](https://github.com/sveltejs/svelte/pull/9482))
- The new `$effect.active` rune returns a boolean to indicate if an effect is active. This lets you set up subscriptions when a value is read (in an effect, or in the template) but also read values without creating subscriptions (**5.0.0-next.10**, [Docs](https://svelte-5-preview.vercel.app/docs/runes#effect-active), [#9591](https://github.com/sveltejs/svelte/pull/9591))
- The new `$effect.root` rune creates a non-tracked scope that doesn't auto-cleanup. This is useful for nested effects that you want to manually control (**5.0.0-next.14**, **Docs**, [#9638](https://github.com/sveltejs/svelte/pull/9638))

> - The latest version of language tools adds best-effort fallback typings to `$props()` rune (**extensions-108.1.0**)

Reviewer 3: it might be helpful to separate changes that are in different projects with a sub-heading or something

- Language tools better supports the `@render` tag by using the `Snippet` type (**extensions-108.0.0**)
- The Svelte ESLint plugin has been updated to support the Svelte 5 preview ([v2.36.0-next.2](https://github.com/sveltejs/eslint-plugin-svelte/releases/tag/v2.36.0-next.2))

> For a complete list of bug fixes and performance updates, check out the [SvelteKit CHANGELOG](https://github.com/sveltejs/kit/blob/master/packages/kit/CHANGELOG.md). You can also find adapter-specific CHANGELOGs in each of [the `adapter` directories](https://github.com/sveltejs/kit/tree/master/packages).

Reviewer 3: Some SvelteKit features:
- https://github.com/sveltejs/kit/commit/3757d3f7d4fa5f489f3e89cb33ec0f59813973f6
- https://github.com/sveltejs/kit/commit/cbd60cc3c787574d60bea5aa3a3119121c6d53e0
- https://github.com/sveltejs/kit/commit/977b151b776a61c1ab7fe3c3ac573b98c0e7e2d5

This one lives in the Kit repo but can be used separately from kit:
- https://github.com/sveltejs/kit/commit/c52aab6e8932420dff6671dc1028fadb8647c24a

I'm also working to get Vite 5 support out the door, so that may happen in the next day or so
